<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="restez">
<title>2. How to search for and fetch sequences • restez</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="2. How to search for and fetch sequences">
<meta property="og:description" content="restez">
<meta property="og:image" content="https://docs.ropensci.org/restez/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">restez</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.1.4.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../articles/restez.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/1_rodents.html">1. Build a database of all rodents</a>
    <a class="dropdown-item" href="../articles/2_search_and_fetch.html">2. How to search for and fetch sequences</a>
    <a class="dropdown-item" href="../articles/3_parsing.html">3. Advanced parsing of a GenBank record</a>
    <a class="dropdown-item" href="../articles/4_phylotar.html">4. Running phylotaR with restez</a>
    <a class="dropdown-item" href="../articles/5_tips_and_tricks.html">5. Tips and Tricks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/restez/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>2. How to search for and fetch sequences</h1>
            
            <h4 data-toc-skip class="date">2023-07-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/restez/blob/HEAD/vignettes/2_search_and_fetch.Rmd" class="external-link"><code>vignettes/2_search_and_fetch.Rmd</code></a></small>
      <div class="d-none name"><code>2_search_and_fetch.Rmd</code></div>
    </div>

    
    
<p>A downside of the <code>restez</code> approach is we are unable to then search the local database for sequences of interest as the database cannot possibly contain sufficient metadata (most notably taxonomic information) to perform as rich a search as online via NCBI. As a result we must perform the sequence discovery process independently from <code>restez</code>. In this tutorial we will demonstrate how to perform online searches using the <code>rentrez</code> package and then use the search results to look up sequences in a <code>restez</code> database.</p>
<p>To run all the code in this tutorial you will need to have already set up the rodents database, see <a href="https://ropensci.github.io/restez/articles/1_rodents.html" class="external-link">Build a database of all rodents</a>.</p>
<div class="section level2">
<h2 id="search-ncbi-genbank-accession-numbers">Search NCBI GenBank accession numbers<a class="anchor" aria-label="anchor" href="#search-ncbi-genbank-accession-numbers"></a>
</h2>
<p>Let’s pretend we’re interested in COI sequences for a group of rodents called the Sciuromorpha. <code>restez</code> provides the <code><a href="../reference/ncbi_acc_get.html">ncbi_acc_get()</a></code> function to make this easy: just provide a text string query <a href="https://www.ncbi.nlm.nih.gov/books/NBK49540/" class="external-link">like you would on NCBI GenBank</a>, and it returns a character vector of accession numbers corresponding to that query. By default, it drops any version number after each accession number. This is not strictly necessary, but it makes it easier to check our results from <code>restez</code> later.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/restez">restez</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 33553 - Sciuromorpha - squirrel-like things</span></span>
<span><span class="va">accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ncbi_acc_get.html">ncbi_acc_get</a></span><span class="op">(</span></span>
<span>  <span class="st">'txid33553[Organism:exp] AND COI [GENE] AND 100:1000[SLEN]'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 492</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">accessions</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "MZ661159" "MZ364430" "HQ966965" "GU670702" "GU670701" "GU670700" "GU670699" "GU670462" "GU670451" "GU670450"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="retrieve-sequences">Retrieve sequences<a class="anchor" aria-label="anchor" href="#retrieve-sequences"></a>
</h2>
<p>To fetch the sequences from the rodents database, we can use the <code>gb_fasta_get</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span><span class="va">rodents_path</span><span class="op">)</span></span>
<span><span class="va">coi_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span><span class="op">(</span>id <span class="op">=</span> <span class="va">accessions</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">coi_sequences</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;  chr "&gt;LT630607.1 Sciurus sp. 1 AG-2016 mitochondrial partial COI gene for cytochrome oxidase subunit 1, specimen vou"| __truncated__</span></span>
<span><span class="co"># Are all accessions in results?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">accessions</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">coi_sequences</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="comparing-to-entrez">Comparing to Entrez<a class="anchor" aria-label="anchor" href="#comparing-to-entrez"></a>
</h2>
<p>Can we not just use <code>rentrez</code> to do the fetching as well? Yes, but <code>restez</code> can be a lot faster. NCBI limits the number of requests per user, often to as little as 100 items per request with varying time delays. Additionally for any programmatic retrieval of sequences using an online server can never be as reliable as a local copy.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># time via restez</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="op">{</span></span>
<span>  <span class="va">coi_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span><span class="op">(</span>id <span class="op">=</span> <span class="va">accessions</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   0.259   0.077   0.272</span></span>
<span><span class="co"># time via Entrez</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="op">{</span></span>
<span>  <span class="va">coi_sequences_p1</span> <span class="op">&lt;-</span> <span class="fu">rentrez</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>,</span>
<span>                                            id <span class="op">=</span> <span class="va">accessions</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>,</span>
<span>                                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span>  <span class="va">coi_sequences_p2</span> <span class="op">&lt;-</span> <span class="fu">rentrez</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>,</span>
<span>                                            id <span class="op">=</span> <span class="va">accessions</span><span class="op">[</span><span class="fl">101</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span>,</span>
<span>                                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span>  <span class="va">coi_sequences_p3</span> <span class="op">&lt;-</span> <span class="fu">rentrez</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>,</span>
<span>                                            id <span class="op">=</span> <span class="va">accessions</span><span class="op">[</span><span class="fl">201</span><span class="op">:</span><span class="fl">300</span><span class="op">]</span>,</span>
<span>                                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span>  <span class="va">coi_sequences_p4</span> <span class="op">&lt;-</span> <span class="fu">rentrez</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>,</span>
<span>                                            id <span class="op">=</span> <span class="va">accessions</span><span class="op">[</span><span class="fl">301</span><span class="op">:</span><span class="fl">400</span><span class="op">]</span>,</span>
<span>                                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span>  <span class="va">coi_sequences_p5</span> <span class="op">&lt;-</span> <span class="fu">rentrez</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>,</span>
<span>                                            id <span class="op">=</span> <span class="va">accessions</span><span class="op">[</span><span class="fl">401</span><span class="op">:</span><span class="fl">456</span><span class="op">]</span>,</span>
<span>                                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   0.050   0.008   5.536</span></span></code></pre></div>
<!-- Below is no longer relevant now that the size of sequences in the db has been limited.
## Missing

A user should know that if an ID cannot be found in the local database no error or warning is raised. This is why it can be good practice to test whether all the provided IDs are in the returned named vector. In this example, we can see that not all the accession IDs that were provided are in the returned `coi_sequences`. Why is that?


```r
# Are all accessions in results?
all(accessions %in% names(coi_sequences))
#> [1] TRUE
# .... no
```

This is because `restez` only downloads GenBank sequences, and these 'missing' sequence IDs are RefSeq sequences. GenBank, however, acts as the source database for RefSeq -- which is the case for other NCBI databases too -- and all the missing RefSeq sequences can also be found in GenBank under a different ID.

We can look up alternative IDs and test whether they are in our `accesssions` vector, so:


```r
(accessions[!accessions %in% names(coi_sequences)])
#> character(0)
# NC* refers to RefSeq sequences and are not available through restez
# The sequence exists in GB under a different id which we can find like so
smmry <- rentrez::entrez_summary(db = 'nucleotide', id = 'NC_027278')
# This ID does exist in our results.
(smmry$assemblyacc %in% accessions)
#> [1] FALSE
```
-->
</div>
<div class="section level2">
<h2 id="next-up">Next up<a class="anchor" aria-label="anchor" href="#next-up"></a>
</h2>
<p><strong><a href="https://docs.ropensci.org/restez/articles/3_parsing.html">Advanced parsing of a GenBank record</a></strong></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
