<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="restez">
<title>Create and Query a Local Copy of GenBank in R • restez</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create and Query a Local Copy of GenBank in R">
<meta property="og:description" content="restez">
<meta property="og:image" content="https://docs.ropensci.org/restez/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">restez</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.1.4.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="active nav-item">
  <a class="nav-link" href="../articles/restez.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/1_rodents.html">1. Build a database of all rodents</a>
    <a class="dropdown-item" href="../articles/2_search_and_fetch.html">2. How to search for and fetch sequences</a>
    <a class="dropdown-item" href="../articles/3_parsing.html">3. Advanced parsing of a GenBank record</a>
    <a class="dropdown-item" href="../articles/4_phylotar.html">4. Running phylotaR with restez</a>
    <a class="dropdown-item" href="../articles/5_tips_and_tricks.html">5. Tips and Tricks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/restez/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Create and Query a Local Copy of GenBank in R</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/restez/blob/HEAD/vignettes/restez.Rmd" class="external-link"><code>vignettes/restez.Rmd</code></a></small>
      <div class="d-none name"><code>restez.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="download-genbank">1. Download GenBank<a class="anchor" aria-label="anchor" href="#download-genbank"></a>
</h2>
<p>In the first section of this introduction to <code>restez</code> we will explain how to download parts of GenBank on to your computer and from these downloads, construct a local database. GenBank can be downloaded from NCBI using the File Transfer Protocol (FTP), the online portal for these downloads can be viewed from your web browser, <a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/">FTP NCBI GenBank</a>. GenBank is hosted at this portal as 1000s of compressed sequence files called flatfiles. New versions of these flatfiles are released once a month as new sequence information is uploaded. For information on the latest GenBank release, see the <a href="https://www.ncbi.nlm.nih.gov/genbank/statistics/" class="external-link">NCBI statistics page</a>.</p>
<p><code>restez</code> downloads a selection of these flatfiles and unpacks them into an SQL-like database (specifically, <a href="https://duckdb.org/" class="external-link">DuckDB</a>) that can be queried using <code>restez</code>’s functions. Because there are potentially huge amounts of sequence information, all of which is unlikely to be of interest to a user, <code>restez</code> allows you to select the sets of flatfiles to download based on GenBank’s domains. For the most part these sets of flatfile types are determined by taxonomy, for example, a user can download all primate or plant sequences.</p>
<p>Here, we will explain how to set up <code>restez</code>, download genbank and then create a database from the downloaded files.</p>
<div class="section level3">
<h3 id="setting-up">1.1 Setting up<a class="anchor" aria-label="anchor" href="#setting-up"></a>
</h3>
<p>Before we can download anything, we need to tell <code>restez</code> where we would like to store the downloaded files and the eventual GenBank database. To do this we use the <code><a href="../reference/restez_path_set.html">restez_path_set()</a></code> function. With this function a user can provide a system file path to a folder within which the function will create a new folder called ‘restez’. In this example, we will create a <code>restez</code> folder in a temporary folder.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set a random number generator seed for reproducibility</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/restez">restez</a></span><span class="op">)</span></span>
<span><span class="co"># rstz_pth &lt;- tempdir()</span></span>
<span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span>filepath <span class="op">=</span> <span class="va">rstz_pth</span><span class="op">)</span></span>
<span><span class="co">#&gt; ... Creating <span style="color: #00BB00;">'/tmp/RtmpA3Wsd0/restez'</span></span></span>
<span><span class="co">#&gt; ... Creating <span style="color: #00BB00;">'/tmp/RtmpA3Wsd0/restez/downloads'</span></span></span></code></pre></div>
<p>A user must always set the <code>restez</code> path in order to use the <code>restez</code> library. Additionally, by running <code><a href="../reference/restez_path_set.html">restez_path_set()</a></code> on a new folder messages are printed to console telling us that the path was created and that a downloads subfolder was created.</p>
<blockquote>
<p><strong>Note</strong>, whenever you intend to use <code>restez</code> you will need to specify the <code>restez</code> path. If you are using it regularly, it might be a good idea to set it using your .Rprofile. <code><a href="../reference/restez_path_set.html">restez_path_set()</a></code> only needs to be run once per R session, but we will use it repeatedly in the chunks below just as a reminder.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="download">1.2 Download<a class="anchor" aria-label="anchor" href="#download"></a>
</h3>
<p>Now we can download GenBank files using <code><a href="../reference/db_download.html">db_download()</a></code>. This is an interactive function that looks up the latest GenBank release, parses the release notes, and prints to console the available sets of flatfiles that can be downloaded. For this example, we will select the smallest available domain which is ‘Unannotated’ and can be pre-specified with ‘20’. Note that the numbering scheme may change between GenBank releases, so you shouldn’t assume a given number will correspond to the same domain if you run it again later. Instead, you can run this function without the <code>preselection</code> argument and wait for the function to prompt you for a domain selection, which will show the current numbering scheme.</p>
<p>After launching, the download function will ask you whether it is OK to download and set up a database for the selected sequence files. If the selection is likely to be large you can always quit the process using either <code>Esc</code> or <code>Ctrl+c</code>. Please note that the diskspace estimates are preliminary. The package tries to be conservative in its estimates.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/db_download.html">db_download</a></span><span class="op">(</span>preselection <span class="op">=</span> <span class="st">'20'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Looking up latest GenBank release ...</span></span>
<span><span class="co">#&gt; ... release number <span style="color: #0000BB;">259</span></span></span>
<span><span class="co">#&gt; ... found <span style="color: #0000BB;">8828</span> sequence files</span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Which sequence file types would you like to download?</span></span>
<span><span class="co">#&gt; Choose from those listed below:</span></span>
<span><span class="co">#&gt; • 1  - <span style="color: #00BB00;">'Invertebrate'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">2099</span> files and <span style="color: #0000BB;">887 GB</span></span></span>
<span><span class="co">#&gt; • 2  - <span style="color: #00BB00;">'Plant (including fungi and algae)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">1713</span> files and <span style="color: #0000BB;">1130 GB</span></span></span>
<span><span class="co">#&gt; • 3  - <span style="color: #00BB00;">'Bacterial'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">1069</span> files and <span style="color: #0000BB;">475 GB</span></span></span>
<span><span class="co">#&gt; • 4  - <span style="color: #00BB00;">'Viral'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">1063</span> files and <span style="color: #0000BB;">471 GB</span></span></span>
<span><span class="co">#&gt; • 5  - <span style="color: #00BB00;">'EST (expressed sequence tag)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">579</span> files and <span style="color: #0000BB;">244 GB</span></span></span>
<span><span class="co">#&gt; • 6  - <span style="color: #00BB00;">'Other vertebrate'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">509</span> files and <span style="color: #0000BB;">216 GB</span></span></span>
<span><span class="co">#&gt; • 7  - <span style="color: #00BB00;">'Rodent'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">314</span> files and <span style="color: #0000BB;">127 GB</span></span></span>
<span><span class="co">#&gt; • 8  - <span style="color: #00BB00;">'Other mammalian'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">273</span> files and <span style="color: #0000BB;">107 GB</span></span></span>
<span><span class="co">#&gt; • 9  - <span style="color: #00BB00;">'GSS (genome survey sequence)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">270</span> files and <span style="color: #0000BB;">117 GB</span></span></span>
<span><span class="co">#&gt; • 10 - <span style="color: #00BB00;">'Patent'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">263</span> files and <span style="color: #0000BB;">112 GB</span></span></span>
<span><span class="co">#&gt; • 11 - <span style="color: #00BB00;">'Constructed'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">238</span> files and <span style="color: #0000BB;">99.7 GB</span></span></span>
<span><span class="co">#&gt; • 12 - <span style="color: #00BB00;">'TSA (transcriptome shotgun assembly)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">127</span> files and <span style="color: #0000BB;">54.2 GB</span></span></span>
<span><span class="co">#&gt; • 13 - <span style="color: #00BB00;">'Environmental sampling'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">95</span> files and <span style="color: #0000BB;">40.7 GB</span></span></span>
<span><span class="co">#&gt; • 14 - <span style="color: #00BB00;">'HTGS (high throughput genomic sequencing)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">82</span> files and <span style="color: #0000BB;">36.8 GB</span></span></span>
<span><span class="co">#&gt; • 15 - <span style="color: #00BB00;">'Primate'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">77</span> files and <span style="color: #0000BB;">31.8 GB</span></span></span>
<span><span class="co">#&gt; • 16 - <span style="color: #00BB00;">'Synthetic and chimeric'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">30</span> files and <span style="color: #0000BB;">11.9 GB</span></span></span>
<span><span class="co">#&gt; • 17 - <span style="color: #00BB00;">'STS (sequence tagged site)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">11</span> files and <span style="color: #0000BB;">4.45 GB</span></span></span>
<span><span class="co">#&gt; • 18 - <span style="color: #00BB00;">'HTC (high throughput cDNA sequencing)'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">8</span> files and <span style="color: #0000BB;">3.49 GB</span></span></span>
<span><span class="co">#&gt; • 19 - <span style="color: #00BB00;">'Phage'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">7</span> files and <span style="color: #0000BB;">3.12 GB</span></span></span>
<span><span class="co">#&gt; • 20 - <span style="color: #00BB00;">'Unannotated'</span></span></span>
<span><span class="co">#&gt;         <span style="color: #0000BB;">1</span> files and <span style="color: #0000BB;">0.00731 GB</span></span></span>
<span><span class="co">#&gt; Provide one or more numbers separated by spaces.</span></span>
<span><span class="co">#&gt; e.g. to download all Mammalian sequences, type: "7 8 15" followed by Enter</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Which files would you like to download?</span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; You've selected a total of <span style="color: #0000BB;">1</span> file(s) and <span style="color: #0000BB;">0.00731 GB</span> of uncompressed data.</span></span>
<span><span class="co">#&gt; These represent: </span></span>
<span><span class="co">#&gt; • <span style="color: #00BB00;">'Unannotated'</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Based on stated GenBank files sizes, we estimate ... </span></span>
<span><span class="co">#&gt; ... <span style="color: #0000BB;">0.00146 GB</span> for  compressed, downloaded files</span></span>
<span><span class="co">#&gt; ... <span style="color: #0000BB;">0.00892 GB</span> for the SQL database</span></span>
<span><span class="co">#&gt; Leading to a total of <span style="color: #0000BB;">0.0104 GB</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co"><span style="font-style: italic;">#&gt; Please note, the real sizes of the database and its downloads cannot be</span></span></span>
<span><span class="co"><span style="font-style: italic;">#&gt; accurately predicted beforehand.</span></span></span>
<span><span class="co"><span style="font-style: italic;">#&gt; These are just estimates, actual sizes may differ by up to a factor of two.</span></span></span>
<span><span class="co"><span style="font-style: italic;">#&gt; </span></span></span>
<span><span class="co">#&gt; Is this OK?</span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Downloading ...</span></span>
<span><span class="co">#&gt; ... <span style="color: #00BB00;">'gbuna1.seq'</span> (<span style="color: #0000BB;">1/1</span>)</span></span>
<span><span class="co">#&gt; Done. Enjoy your day.</span></span></code></pre></div>
<p>Now the download has begun, we just need to wait for it to finish. For the largest domains, this can take quite a while and may be prone to server errors. In this case, you may want to set the <code>max_tries</code> argument of <code><a href="../reference/db_download.html">db_download()</a></code> to a relatively high number to automatically re-try downloading until that number of tries is exceeded (in this case, be sure to set <code>overwrite</code> to <code>FALSE</code> or the download will start over from scratch).</p>
</div>
<div class="section level3">
<h3 id="create-a-database">1.3 Create a database<a class="anchor" aria-label="anchor" href="#create-a-database"></a>
</h3>
<p>After the download has completed, we need to use <code><a href="../reference/db_create.html">db_create()</a></code> to create our local database. This looks in the downloads folder of our <code>restez</code> path, breaks these files up into separate GenBank records and adds them to the SQL-like database. Again, for very large collections of sequences this can take quite a while.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/db_create.html">db_create</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Inspecting <span style="color: #0000BB;">1</span> file(s) to add to the database ...</span></span>
<span><span class="co">#&gt; ... <span style="color: #00BB00;">'gbuna1.seq.gz'</span> (<span style="color: #0000BB;">1/1</span>)</span></span>
<span><span class="co">#&gt; Done.</span></span></code></pre></div>
<p><code><a href="../reference/db_create.html">db_create()</a></code> allows a user to specify minimum and maximum sequence lengths. It’s always a good idea to limit the number of sequences in a database so that look-up times are faster. If you know you are only interested in certain lengths of sequences it is a good idea to limit the sequences in the database at this stage. This can also be done with the <code>acc_filter</code> argument, as described in the <a href="https://docs.ropensci.org/restez/articles/5_tips_and_tricks.html">“Tips and Tricks” vignette</a>. You can always run <code><a href="../reference/db_create.html">db_create()</a></code> again to change the limits. You will simply need to delete the original database first with <code><a href="../reference/db_delete.html">db_delete()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="checking-the-setup">1.4 Checking the setup<a class="anchor" aria-label="anchor" href="#checking-the-setup"></a>
</h3>
<p>After the download and the database creation steps are complete, we can confirm our setup using <code><a href="../reference/restez_status.html">restez_status()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span><span class="va">rstz_pth</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/restez_status.html">restez_status</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Checking setup status at  ...</span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Restez path ...</span></span>
<span><span class="co">#&gt; ... Path <span style="color: #00BB00;">'/tmp/RtmpA3Wsd0/restez'</span></span></span>
<span><span class="co">#&gt; ... Does path exist? <span style="color: #00BB00;">'Yes'</span></span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Download ...</span></span>
<span><span class="co">#&gt; ... Path <span style="color: #00BB00;">'/tmp/RtmpA3Wsd0/restez/downloads'</span></span></span>
<span><span class="co">#&gt; ... Does path exist? <span style="color: #00BB00;">'Yes'</span></span></span>
<span><span class="co">#&gt; ... N. files <span style="color: #0000BB;">2</span></span></span>
<span><span class="co">#&gt; ... Total size <span style="color: #0000BB;">3.08M</span></span></span>
<span><span class="co">#&gt; ... GenBank division selections <span style="color: #00BB00;">'Unannotated'</span></span></span>
<span><span class="co">#&gt; ... GenBank Release <span style="color: #0000BB;">259</span></span></span>
<span><span class="co">#&gt; ... Last updated <span style="color: #00BB00;">'2024-03-23 06:25:54.376743'</span></span></span>
<span><span class="co">#&gt; ────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Database ...</span></span>
<span><span class="co">#&gt; ... Path <span style="color: #00BB00;">'/tmp/RtmpA3Wsd0/restez/sql_db'</span></span></span>
<span><span class="co">#&gt; ... Does path exist? <span style="color: #00BB00;">'Yes'</span></span></span>
<span><span class="co">#&gt; ... Total size <span style="color: #0000BB;">8.76M</span></span></span>
<span><span class="co">#&gt; ... Does the database have data? <span style="color: #00BB00;">'Yes'</span></span></span>
<span><span class="co">#&gt; ... Number of sequences <span style="color: #0000BB;">768</span></span></span>
<span><span class="co">#&gt; ... Min. sequence length <span style="color: #0000BB;">0</span></span></span>
<span><span class="co">#&gt; ... Max. sequence length <span style="color: #0000BB;">Inf</span></span></span>
<span><span class="co">#&gt; ... Last_updated <span style="color: #00BB00;">'2024-03-23 06:25:55.741279'</span></span></span></code></pre></div>
<p>The status function allows a user to always touch base with their <code>restez</code> set-up. It can be run at any point, before and/or after download or database creation. It is also designed to provide useful messages to a user for what they need to do next in order for them to make queries to their database. If you ever get confused, run <code><a href="../reference/restez_status.html">restez_status()</a></code> to see what is happening.</p>
<p>Additionally, if you are developing your own functions that depend on <code>restez</code>, you can use <code><a href="../reference/restez_ready.html">restez_ready()</a></code> which will simply return TRUE or FALSE on whether the database can be queried.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span><span class="va">rstz_pth</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="../reference/restez_ready.html">restez_ready</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">'Database is ready to be queried!'</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">'Database cannot be queried :-('</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; [1] "Database is ready to be queried!"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="query">2. Query<a class="anchor" aria-label="anchor" href="#query"></a>
</h2>
<p>Once a restez database has been set up we can query the database using the <code>gb_*_get()</code> functions. These functions allow us to retrieve specific columns in the SQL-like database: ‘sequence’, ‘definition’, ‘accession’, ‘version’ and ‘organism’. Also, they allow us to get the whole text formatted record and sequence data in fasta format. In this example, we can use <code><a href="../reference/list_db_ids.html">list_db_ids()</a></code> to identify accession numbers in the database. We could also use <code>entrez_search()</code> provided the database contains sequences of interest to us, see <a href="https://docs.ropensci.org/restez/articles/2_search_and_fetch.html">‘search and fetch’</a>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span><span class="va">rstz_pth</span><span class="op">)</span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="../reference/list_db_ids.html">list_db_ids</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>, n <span class="op">=</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">ids</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "AH000952"</span></span>
<span><span class="co"># sequence</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/gb_sequence_get.html">gb_sequence_get</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named chr "GGGGTCTGACGCTCAGTGGAACGAAAACTCACGTTAAGGGATTTTGGTCATGAGATTATCAAAAAGGATCTTCACCTAGATCCTTTTAAATTAAAAATGAAGTTTTAAATC"| __truncated__</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr "AH000952"</span></span>
<span><span class="co"># definition</span></span>
<span><span class="op">(</span><span class="fu"><a href="../reference/gb_definition_get.html">gb_definition_get</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                                                AH000952 </span></span>
<span><span class="co">#&gt; "Transposon Tn3 transposon Tn3 ampicillin resistance (amp-r) product gene, partial cds"</span></span>
<span><span class="co"># version</span></span>
<span><span class="op">(</span><span class="fu"><a href="../reference/gb_version_get.html">gb_version_get</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;     AH000952 </span></span>
<span><span class="co">#&gt; "AH000952.2"</span></span>
<span><span class="co"># organism</span></span>
<span><span class="op">(</span><span class="fu"><a href="../reference/gb_organism_get.html">gb_organism_get</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;         AH000952 </span></span>
<span><span class="co">#&gt; "Transposon Tn3"</span></span>
<span><span class="co"># fasta</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; &gt;AH000952.2 Transposon Tn3 transposon Tn3 ampicillin resistance (amp-r) product gene, partial cds</span></span>
<span><span class="co">#&gt; GGGGTCTGACGCTCAGTGGAACGAAAACTCACGTTAAGGGATTTTGGTCATGAGATTATCAAAAAGGATC</span></span>
<span><span class="co">#&gt; TTCACCTAGATCCTTTTAAATTAAAAATGAAGTTTTAAATCAATCTAAAGTATATATGAGTAAACTTGGT</span></span>
<span><span class="co">#&gt; CTGACAGTTACCAATGCTTAATCAGTGAGGCACCTATCTCAGCGATCTGTCTATTTCGTTCATCCATAGT</span></span>
<span><span class="co">#&gt; TGCCTGACTCCCCGTCGTGTAGATAACTACGATACGGGAGGGCTTACCATCTGGCCNNNNNNNNNNNNNN</span></span>
<span><span class="co">#&gt; NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span></span>
<span><span class="co">#&gt; NNNNNNNNNNNNNNNNGGCCATTATTCCTTCACGCTGGCAGAACTGGTGACCAAAGGTCATCTGAGACCA</span></span>
<span><span class="co">#&gt; TTAAAAGAGGCGTCAGAGGCAGAAAACGTTGCTTAACGTGAGTTTTCGTTCCACTGAGCGTCAGACCCC</span></span>
<span><span class="co"># Note, for large databases these requests can take a long time to complete.</span></span>
<span><span class="co"># Always try and limit the size of the database by specifying min and max</span></span>
<span><span class="co"># sequence lengths with db_create()</span></span>
<span><span class="co"># Also note, if an id is not present in the database nothing is returned</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">id</span>, <span class="st">'notanid'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; &gt;AH000952.2 Transposon Tn3 transposon Tn3 ampicillin resistance (amp-r) product gene, partial cds</span></span>
<span><span class="co">#&gt; GGGGTCTGACGCTCAGTGGAACGAAAACTCACGTTAAGGGATTTTGGTCATGAGATTATCAAAAAGGATC</span></span>
<span><span class="co">#&gt; TTCACCTAGATCCTTTTAAATTAAAAATGAAGTTTTAAATCAATCTAAAGTATATATGAGTAAACTTGGT</span></span>
<span><span class="co">#&gt; CTGACAGTTACCAATGCTTAATCAGTGAGGCACCTATCTCAGCGATCTGTCTATTTCGTTCATCCATAGT</span></span>
<span><span class="co">#&gt; TGCCTGACTCCCCGTCGTGTAGATAACTACGATACGGGAGGGCTTACCATCTGGCCNNNNNNNNNNNNNN</span></span>
<span><span class="co">#&gt; NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span></span>
<span><span class="co">#&gt; NNNNNNNNNNNNNNNNGGCCATTATTCCTTCACGCTGGCAGAACTGGTGACCAAAGGTCATCTGAGACCA</span></span>
<span><span class="co">#&gt; TTAAAAGAGGCGTCAGAGGCAGAAAACGTTGCTTAACGTGAGTTTTCGTTCCACTGAGCGTCAGACCCC</span></span></code></pre></div>
<p>Additionally, for more flexibility and options for extracting sequence record information see <a href="https://docs.ropensci.org/restez/articles/3_parsing.html">GenBank record parsing</a>.</p>
</div>
<div class="section level2">
<h2 id="entrez">3. Entrez<a class="anchor" aria-label="anchor" href="#entrez"></a>
</h2>
<p>Entrez wrappers are part of the <code>restez</code> package. These allow a user to make use of the local GenBank using functions that were built for <a href="https://github.com/ropensci/rentrez" class="external-link"><code>rentrez</code></a>. This minimizes the amount of coding changes required for any Entrez dependent code.</p>
<blockquote>
<p>Currently, only <code><a href="../reference/entrez_fetch.html">entrez_fetch()</a></code> is available with restez and only text formatted rettypes are allowed.</p>
</blockquote>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/restez_path_set.html">restez_path_set</a></span><span class="op">(</span><span class="va">rstz_pth</span><span class="op">)</span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="../reference/list_db_ids.html">list_db_ids</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>, n <span class="op">=</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">ids</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "AF298087"</span></span>
<span><span class="co"># get fasta sequences with entrez</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">restez</span><span class="fu">::</span><span class="fu"><a href="../reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>, id <span class="op">=</span> <span class="va">id</span>, rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; &gt;AF298087.1 Unidentified clone A5 DNA sequence from ocean beach sand</span></span>
<span><span class="co">#&gt; GATCCGGTCNTCGCGCTTTGCGCCTCGCGGATCTCGCGCACCTGCAGCTGGGTCACCAGCGGCGAGATGA</span></span>
<span><span class="co">#&gt; TGGTGCCTAGCAGCAGCCCGAAGACCACCATCACCACCAGGATCTCGATCAGGGTGAAGCCCTGGTCAGG</span></span>
<span><span class="co">#&gt; ACGCCGGCTCACGGCGCGGGCGGTGGCTTCGCGGCGCATCACAGGCACGGCCTTGCTGGCGTGGTTGGGG</span></span>
<span><span class="co">#&gt; GCGGCACTGGATTATTGCAGGCCACCATTTCCAGGTAATCGTTAAAGCCCATCTTACGTGCCGTGCGCTC</span></span>
<span><span class="co">#&gt; CATGAGCAGGTCGACACCATTCGCGTTGTCTTCCTCGAAGCCGGCCTGGCCGGGAGTCGCCAGGGTGAGG</span></span>
<span><span class="co">#&gt; TCCTGCGTTGGCGCAAGCTCACCGCTCACGATCACCACCGCGTGGGCATCGTCCACGACTGAAGCCCCAT</span></span>
<span><span class="co">#&gt; CGGCGACGTTGACGACGACGCANTCGCCTCCGACGGTGCAGGGGTCGGACGGGCCCCCGGCTCCNGTGGG</span></span>
<span><span class="co">#&gt; GCTGAAGCCGTTGGCGTANCTCGCATANACAAACTGNGCCCACCGTTCTCGGNGAACCANGCGGGAGATC</span></span>
<span><span class="co">#&gt; CCAGGGTTTCCGGGCCTG</span></span>
<span><span class="co"># entrez_fetch will also search via Entrez for any ids not in the db</span></span>
<span><span class="va">plant_sequence</span> <span class="op">&lt;-</span> <span class="st">'AY952423'</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">restez</span><span class="fu">::</span><span class="fu"><a href="../reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">'nucleotide'</span>, id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">id</span>, <span class="va">plant_sequence</span><span class="op">)</span>,</span>
<span>                            rettype <span class="op">=</span> <span class="st">'fasta'</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] id(s) are unavailable locally, searching online.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; &gt;AF298087.1 Unidentified clone A5 DNA sequence from ocean beach sand</span></span>
<span><span class="co">#&gt; GATCCGGTCNTCGCGCTTTGCGCCTCGCGGATCTCGCGCACCTGCAGCTGGGTCACCAGCGGCGAGATGA</span></span>
<span><span class="co">#&gt; TGGTGCCTAGCAGCAGCCCGAAGACCACCATCACCACCAGGATCTCGATCAGGGTGAAGCCCTGGTCAGG</span></span>
<span><span class="co">#&gt; ACGCCGGCTCACGGCGCGGGCGGTGGCTTCGCGGCGCATCACAGGCACGGCCTTGCTGGCGTGGTTGGGG</span></span>
<span><span class="co">#&gt; GCGGCACTGGATTATTGCAGGCCACCATTTCCAGGTAATCGTTAAAGCCCATCTTACGTGCCGTGCGCTC</span></span>
<span><span class="co">#&gt; CATGAGCAGGTCGACACCATTCGCGTTGTCTTCCTCGAAGCCGGCCTGGCCGGGAGTCGCCAGGGTGAGG</span></span>
<span><span class="co">#&gt; TCCTGCGTTGGCGCAAGCTCACCGCTCACGATCACCACCGCGTGGGCATCGTCCACGACTGAAGCCCCAT</span></span>
<span><span class="co">#&gt; CGGCGACGTTGACGACGACGCANTCGCCTCCGACGGTGCAGGGGTCGGACGGGCCCCCGGCTCCNGTGGG</span></span>
<span><span class="co">#&gt; GCTGAAGCCGTTGGCGTANCTCGCATANACAAACTGNGCCCACCGTTCTCGGNGAACCANGCGGGAGATC</span></span>
<span><span class="co">#&gt; CCAGGGTTTCCGGGCCTG</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; &gt;AY952423.1 Livistona chinensis tRNA-Lys (trnK) gene, partial sequence; and matK gene, complete sequence; chloroplast</span></span>
<span><span class="co">#&gt; ATTGGGGTTGCTAACTCAACGGTAGAGTACTCGGCTTTTAAGTGCGACTATCATCTTTTACACATTTGGA</span></span>
<span><span class="co">#&gt; TGAAGTAAGGAATTCGTCCAGACTATTGGTAGAGTCTATAAGACCACGACTGATCCTGAAAGGTAATGAA</span></span>
<span><span class="co">#&gt; TGGAAAAAATAGCATGTCGTACGTAATACAATGAGAAACTTGTAATTTCTTATTGTAATTTTTTAAGTAG</span></span>
<span><span class="co">#&gt; AACTTTGAGTTTATCCTTACTGGATCATTACAAAAATATTGTATTTTATTTTTGGAAGGGGACGAAAAAA</span></span>
<span><span class="co">#&gt; AGGAAATTCCCAACATTTATTGTTTGGTCTAATGAATAAATGGATAGGGGCCTAGGGTAGGGCCCAATTT</span></span>
<span><span class="co">#&gt; TTGTAAAACAAAAAGCAACGAGCTTATGTTCTTAATTTGAATAATTACCCGATCTAATTAGATGTTAAAA</span></span>
<span><span class="co">#&gt; ATAAATTAGTGCCAGATGTGGTAAAGGGTTCTACTGTAAGTGGACCTTTTTTTTTTTTTTTTATGAATCC</span></span>
<span><span class="co">#&gt; TACCTATTATCTATTATGGATTAAAGATGGATGTGTATAAGAAGAAGTATACTGATAAAGAGAATTTTTC</span></span>
<span><span class="co">#&gt; CAAAGTCAAAAGAGCAATCGGGTTGCAAAAATAAAGGATTTTTACCTCCGAGAATTATAAATTAATTGGA</span></span>
<span><span class="co">#&gt; TCAAAAGGAGAGGAAAAAGTCTGTGATTGGACTCCTTCTATCCGCGGGTATGGGTATATAGTAGGTATAT</span></span>
<span><span class="co">#&gt; ATGTATATTTGTATACTATATAAATTACATGCCCTGTTCTGACCGTATTGCACTATGTATTATTTGATAA</span></span>
<span><span class="co">#&gt; TCCAAGAAATGCCTCCTACTTCTGGTTCAAGTAGAAATGAAAATAGAAGAATTACAAGAATATTTAGAAA</span></span>
<span><span class="co">#&gt; AAGATAGATCTCGGCAACAACACTTCCTATACCCACTTTTCTTTCAGGAGTATATTTATGCACTTGCTCA</span></span>
<span><span class="co">#&gt; TGATTATGGGTTTAAAGGGTTCGATTTTTTACGAACCTATGGAAATTGGGGGTTATGATAATAAATCTAG</span></span>
<span><span class="co">#&gt; TTCAGTACTTGTAAAACATTTAATTACTCGAATGTATCAACAGAATTATTTGATTTATTCTGTTAATGAA</span></span>
<span><span class="co">#&gt; TCTAACCAAAATCGATTGATTGAGCATAACAATTCTTTTTATTCTCAAATGATATCTGAAGTTTTTGCGA</span></span>
<span><span class="co">#&gt; TCATTGCAGAAATTCCATTCTCTCAGCAATTACTATTTTCTCTTCGAGGAAAAAAGAATACCAAAATCTC</span></span>
<span><span class="co">#&gt; AGACTTTACGATCTATTCATTCAATATTTCCCTTTTTAGAAGACAAATTATCACATTTAAACTATGTGTC</span></span>
<span><span class="co">#&gt; AGATATATTAATACCCTATCCCATCCATTTGGAAATCTTGGTGCAAATTCTTCAATGCTGGATCCAAGAT</span></span>
<span><span class="co">#&gt; GTTTCTTCTTTGCATTTATTGCGATTCTTTCTCCACGAACATCATAATGGGAATAGTTTTTTTTTTCCAA</span></span>
<span><span class="co">#&gt; AGAAATCCTTTTCAAAAGAAAATAAAAGACTCTTTCGATTCCTATATAATTCTTATGTATCTGAATGTGA</span></span>
<span><span class="co">#&gt; ATTTGTCTTAGTGTTTCTTCGTAAACAATCCTCTTATTTACAATCAAAATCCTATGGAATCTTTCTTGAG</span></span>
<span><span class="co">#&gt; CGAACACATTTCTATGGAAGAATGGAACATCTTATAGTAGTGTGTCATAATTATTGTCAGAAGGCCTTTT</span></span>
<span><span class="co">#&gt; GGGTCTTCAAGGATCCTTTTATGCATTATGTTCGATATCAAGGAAAAGCAATTCTGGCATCAAAAGGATC</span></span>
<span><span class="co">#&gt; TTATCTTTTGATGAAGAAATGGAGATGTCATCTTGTCAATTTCTGGCAATATTATTTTCATTTTTGGGCT</span></span>
<span><span class="co">#&gt; CAGCCTTACAGAATTTCAATAAACCAATTAGGAAATCATTCCTTCTATTTTCTCGGTTATCTTTCAAGTG</span></span>
<span><span class="co">#&gt; TATTAAAAAATACTTCGTCTGTAAGGAATCAAATGCTAGAGAATTCCTTTTTAATAGATACTATTACTAA</span></span>
<span><span class="co">#&gt; TAAATTGGATACCATAGTCCCAGTTCTTCCTCTTATTGGATCTTTGTCTAAAGCTAAATTTTGTACCGTA</span></span>
<span><span class="co">#&gt; TCCGGGCATCCTAGTAGTAAGCCAATCTGGACGGATTTATCGGATTCTGATATTATTGATAGATTTGGTC</span></span>
<span><span class="co">#&gt; GGATATGTAGAAATCTTTCTCATTATTATAGTGGATCCTCAAAAAAACAGAGCTTATATCGAATAAGGTA</span></span>
<span><span class="co">#&gt; TATACTTCGACTTTCTTGTGCTAGAACTTTAGCTCGTAAACATAAAAGTACAGTACGTGCTTTTTTGCAA</span></span>
<span><span class="co">#&gt; AGATTAGGTTCGGAATTATTAGAAGAATTCTTTACAGAAGAAGAAGGAGTTGTTTTTTTGATTTCCCAAA</span></span>
<span><span class="co">#&gt; AGAACAAAACCTCTTTTCCTCTCTATAGGTCACATAGAGAACGCATTTGGTATTTGGATATTATCCATAT</span></span>
<span><span class="co">#&gt; TAATGAATTGGTGAATTCATTTATGATGGGGCGATAAGCCCCTATAAAATAAGAAATATAAATTTTTTCT</span></span>
<span><span class="co">#&gt; AATGTCTAATAAATAGACGACAAATTCATTAATTTTCATTCTGAAATGCTCATCTAGTAGTGTAGTGATT</span></span>
<span><span class="co">#&gt; GAATCAACTGAGTATTCAAAATTTTTAGACAAACTTCTAGGGATAGAAGTTTGTTTTATCTGTATACATA</span></span>
<span><span class="co">#&gt; GGTAAAGTCGTGTGCAATGAAAAATGCAAGCACGATTTGGGGAGAGATAATTTTCTCTATTGTAACAAAT</span></span>
<span><span class="co">#&gt; AAAAATTATCTACTCCATCCGACTAGTTAATCG</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="more-information">4. More information<a class="anchor" aria-label="anchor" href="#more-information"></a>
</h2>
<p>For more information about <code>restez</code> see the other tutorials:</p>
<ol style="list-style-type: decimal">
<li><a href="https://docs.ropensci.org/restez/articles/1_rodents.html">Build a database of all rodents</a></li>
<li><a href="https://docs.ropensci.org/restez/articles/2_search_and_fetch.html">How to search for and fetch sequences</a></li>
<li><a href="https://docs.ropensci.org/restez/articles/3_parsing.html">Advanced parsing of a GenBank record</a></li>
<li><a href="https://docs.ropensci.org/restez/articles/4_phylotar.html">Running phylotaR with restez</a></li>
<li><a href="https://docs.ropensci.org/restez/articles/5_tips_and_tricks.html">Tips and tricks</a></li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
